### PR TITLE
ENH: Updating the sobel CBH retrieval method to be more robust for ot…

### PR DESCRIPTION
**Pull request by AdamTheisen**
…her instruments

- [x] Tests added
- [x] Documentation reflects changes
- [x] PEP8 Standards or use of linter
- [x] Xarray Dataset or DataArray variable naming follows 'ds' or 'da' naming

### CODE REVIEW

**Review by zssherman — APPROVED** [review]
This looks good to me @AdamTheisen!

**Comment by mgrover1** [comment]
Boolean indexes are not allowed with dask arrays since dask is not aware of the actual values until they are loaded... for that failing test, `.values` needs to be added after 

```python
ds['alt']
```

**Comment by zssherman** [comment]
@mgrover1 Ah makes sense!

**Comment by AdamTheisen** [comment]
Thanks @mgrover1 !  I just tested that on my side and works.

File: act/retrievals/cbh.py [hunk 1]
```diff
@@ -34,7 +39,13 @@ def generic_sobel_cbh(
     var_thresh : float
         Thresholding for variable if needed.
     fill_na : float
-        What to fill nans with in DataArray if any.
+        Value to fill nans with in DataArray if any.
+    uniform_filter : boolean
```
Comment (mgrover1) [inline act/retrievals/cbh.py:43]: Can you rename this to `filter_type` and set it to `uniform`? That way in the future, if we want to add additional filter types, we can?

**Review by mgrover1 — CHANGES_REQUESTED** [review]
A couple of comments and suggestions!

File: act/retrievals/cbh.py (hunk 1 above)
Comment (AdamTheisen) [reply act/retrievals/cbh.py]: Done!

File: act/retrievals/cbh.py [hunk 2]
```diff
@@ -79,42 +90,46 @@ def generic_sobel_cbh(
...
     # Get height variable to use for cbh
     height = ds[height_dim].values
 
     # Run through times and find the height
     cbh = []
     for i in range(np.shape(diff)[0]):
-        index = np.where(diff[i, :] > 5.0)[0]
+        try:
+            index = np.where(diff[i, :] > edge_thresh)[0]
+        except:
```
[the remark this answers is not in the document]
Comment (AdamTheisen) [reply act/retrievals/cbh.py]: Done!

**Comment by AdamTheisen** [comment]
I have a feeling we might have more errors with these xarray where/boolean issues.

**Review by mgrover1 — APPROVED** [review]

**Comment by AdamTheisen** [comment]
Great, if all passes and approvals are there, I'll merge on in!

**Pull request merged**